File: src/main/java/com/cfgglobal/test/dao/BranchDao.java
package com.cfgglobal.test.dao;

import com.cfgglobal.test.dao.base.BaseDao;
import com.cfgglobal.test.domain.Branch;
import io.vavr.collection.List;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;
import org.springframework.stereotype.Repository;

@Repository
public interface BranchDao extends BaseDao<Branch, Long> {


    @Query("SELECT id from Branch where parent_id = :id ")
    List<Long> findSubOrgIds(@Param("id") Long id);

}

File: src/main/kotlin/com/cfgglobal/test/dao/BranchDao.kt
package com.cfgglobal.test.dao

import com.cfgglobal.test.dao.base.BaseDao
import com.cfgglobal.test.domain.Branch

import org.springframework.data.jpa.repository.Query
import org.springframework.data.repository.query.Param
import org.springframework.stereotype.Repository

@Repository
interface BranchDao : BaseDao<Branch, Long> {


    @Query("SELECT id from Branch where parent_id = :id ")
    fun findSubOrgIds(@Param("id") id: Long?): List<Long>

}